index.js

```diff
@@ -14,16 +35,20 @@ module.exports = function requireFromString(code, filename, opts) {
 	opts.appendPaths = opts.appendPaths || [];
 	opts.prependPaths = opts.prependPaths || [];
 
-	if (typeof code !== 'string') {
-		throw new Error('code must be a string, not ' + typeof code);
-	}
-
 	var paths = Module._nodeModulePaths(path.dirname(filename));
 
 	var m = new Module(filename, module.parent);
+
+	var requireHook = opts.require;
```

Comment:
yep, for sure.

Still a work in progress. (which I now realize I did not mention).